library/alloc/src/vec/mod.rs

```diff
@@ -3259,6 +3259,65 @@ impl<T, A: Allocator> Vec<T, A> {
         // - `cap / N` fits the size of the allocated memory after shrinking
         unsafe { Vec::from_raw_parts_in(ptr.cast(), len / N, cap / N, alloc) }
     }
+
+    /// This clears out this `Vec` and recycles the allocation into a new `Vec`.
+    /// The item type of the resulting `Vec` needs to have the same size and
+    /// alignment as the item type of the original `Vec`.
+    ///
+    /// # Examples
+    ///
+    ///  ```
+    /// #![feature(vec_recycle)]
+    /// #![feature(transmutability)]
+    /// let a: Vec<u8> = vec![0; 100];
+    /// let capacity = a.capacity();
+    /// let addr = a.as_ptr().addr();
+    /// let b: Vec<i8> = a.recycle();
+    /// assert_eq!(b.len(), 0);
+    /// assert_eq!(b.capacity(), capacity);
+    /// assert_eq!(b.as_ptr().addr(), addr);
+    /// ```
+    ///
+    /// ```ignore // FIXME(transmutability): https://github.com/rust-lang/rust/issues/148227#issuecomment-3478099591
```

Comment:
This fixes the doc-tests:

```suggestion
    /// ```ignore
    /// // FIXME(transmutability): https://github.com/rust-lang/rust/issues/148227#issuecomment-3478099591
```